### PR TITLE
SCAL-74116-docs-deployment-on-github-pages

### PR DESCRIPTION
**Pull request by pankajp-ts**
Ticket - https://thoughtspot.atlassian.net/browse/SCAL-74116

- did GitHub pages setup on repo
- added deployment script

### CODE REVIEW

**Review by ayonghosh — CHANGES_REQUESTED** [review]
Let's set it up such that we can deploy Github pages from any Git branch

File: package.json [hunk 1]
```diff
@@ -11,7 +11,9 @@
         "build": "rollup -c",
         "watch": "rollup -cw",
         "develop-docs": "gatsby develop",
-        "build-docs": "gatsby build",
+        "build-docs": "gatsby build --prefix-paths",
+        "deploy-docs": "gatsby build --prefix-paths && gh-pages -d public -b 7.3.0-docs",
+        "publish-docs": "gh-pages -d public -b 7.3.0-docs",
```
Comment (ayonghosh) [inline package.json:16]: We need to parameterise the branch name. We can't hardcode 7.3.0-docs as there could be many branches in future.

File: gatsby-config.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 module.exports = {
+    pathPrefix: '/embed-sdk',
```
Comment (ayonghosh) [inline gatsby-config.js:2]: Do we really need the pathPrefix?

**Review by ayonghosh — CHANGES_REQUESTED** [review]

File: gatsby-config.js (hunk 2 above)
Comment (pankajp-ts) [reply gatsby-config.js:2]: yes it's needed since this is a organization and we have multiple repos in it

File: scripts/gatsby-commands.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+                        `Invalid value (ex.: option=value) passed to '${cmdOptions[0]}' option: `,
+                        cmdOptions[1],
+                    );
+                } else {
+                    const branchName = cmdOptions[1];
+                    return branchName;
+                }
+            } else {
+                // default branch name for publish command
+                return '7.3.0-docs';
```
Comment (ayonghosh) [inline scripts/gatsby-commands.js:55]: Let's not use any default value. Tell the user than a branch name needs to be specified explicitly.

**Review by ayonghosh — COMMENTED** [review]
LGTM after addressing comment

File: scripts/gatsby-commands.js (hunk 3 above)
Comment (pankajp-ts) [reply scripts/gatsby-commands.js:55]: but same branch needs to be set in settings as well, so this is two way process.

Comment (pankajp-ts) [reply scripts/gatsby-commands.js:55]: also whatever branch name you will specify here will loose all the files and folder except what is generated by build inside public folder

**Pull request merged**